proposals/0000-uncurated-layer.rst

```diff
@@ -0,0 +1,53 @@
+.. proposal-number:: Leave blank. This will be filled in when the proposal is
+                     accepted.
+
+.. highlight:: haskell
+
+Uncurated Hackage Layer
+==============
+
+Motivation
+----------
+
+There is a tension between two purposes of Hackage -- first as a central repository of Haskell code, and second as a curated store that has artifacts that are intended to be correctly built and depended upon in a self-contained fashion. (I.e. which contain all information necessary to correctly build them). The way the latter is accomplished is by asking packages follow the package version policy (https://pvp.haskell.org/) that is used to inform clients of a package of changes to that package that might affect them, and to provide a way for clients to specify a particular range of versions of a dependency that they are compatible with.
+
+The aim of this proposal is to separate these two purposes, by allowing authors to distinguish if they wish to opt-out of following the pvp and the attendant curation process that helps to maintain correct dependency information. In so doing, they recognize that their discoverability on Hackage may be reduced, since the UI will give precedence to those packages which, in conjunction with potential cabal revisions, are self-contained with regards to the information necessary to build them.
+
+Proposed Change
+---------------
+
+This is a phased proposal with gradual rollout possible, in dependency order of the steps. The first phase is immediate. Other phases may be rolled out at any pace, but necessarily in the order specified (including the collapse of two steps into a joint step).
+
+The desired end-state will have the following properties:
+
+1) Packages will have an additional flag set in the Hackage package database, that indicates if they are curated or not. This flag is set *per version*.
+2) Package authors will set this flag *on upload*, by setting the "x-uncurated" property of the cabal file of a package to "true". If no "x-uncurated" property is set, this will be considered "false".
```

Comment:
IMHO it is better to avoid additional negations when naming booleans, especially since you're putting in a sensible default anyway. (Image code like `if not (uncurated p)`).

Instead just call it `x-curated`, default `true`?

Reply:
My thought was that unset flags by convention default to "false". But this is a good argument the other way. Don't have a strong opinion here.

Reply:
I also prefer to avoid double negations.

Reply:
In the spirit of avoiding boolean blindness can we please call the flag `x-curation` and its values `curated` and `uncurated`?